sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2244,6 +2244,15 @@ object GpuOverrides {
         ("map", TypeSig.MAP.nested(TypeSig.STRING), TypeSig.MAP.nested(TypeSig.all)),
         ("key", TypeSig.lit(TypeEnum.STRING), TypeSig.all)),
       (in, conf, p, r) => new GpuGetMapValueMeta(in, conf, p, r)),
+    expr[ElementAt](
+      "Returns element of array at given(1-based) index in value if column is array. " +
+        "Returns value for the given key in value if column is map.",
+      ExprChecks.binaryProjectNotLambda(
+        TypeSig.commonCudfTypes, TypeSig.all,
+        ("left", TypeSig.ARRAY.nested(TypeSig.commonCudfTypes) +
```

Comment:
Type checking is very simple with how it handles nested types. They all end up being combined and this would allow not just arrays of any of the common cudf types, but also maps.

If we want to support all of them we are going to need extra checks that if it is a map then it is only for the types that we support, and we might want to put some notes on them too.

Reply:
Added extra type check in `tagExprForGpu`, also removed the class for meta, moved it under the type check part directly to make it more intuitive.